index.d.ts

```diff
@@ -11,12 +11,9 @@ declare namespace FirstChunkStream {
 		readonly chunkLength: number;
 	}
 
-	type TransformFunction = (
-		error: Error | null,
-		chunk: Buffer,
-		encoding: string,
-		callback: (error?: Error | null, buffer?: string | Buffer | Uint8Array, encoding?: string) => void
-	) => void;
+	type NullableBufferLike = string | Buffer | Uint8Array | void | null;
```

Comment:
I don't really like using `null` to indicate that the stream should end. It feels way too magic and can happen accidentally. I think we should use a Symbol for this instead. Similar to https://github.com/sindresorhus/find-up/blob/a2841237562d1f9996084d16c1bc6b471143892a/index.d.ts#L8

Reply:
It is not ending though, it is only skipping first chunk. What name you propose for such symbol?

Reply:
Actually, zero length buffer or string are the same as skipping, so why not just prohibit `null`/`undefined`?

Reply:
> It is not ending though, it is only skipping first chunk.

I'm pretty sure it is. `null` is special with streams and causes stream EOF. https://stackoverflow.com/a/36761144

> Actually, zero length buffer or string are the same as skipping, so why not just prohibit null/undefined?

Yes, that's what I'm proposing, but there still needs to be a way to end the stream.

Reply:
It does
```
if (!buffer) {
	done();
	return;
}
```

so null won't ever make it to push queue.

Reply:
Don't you want to call the symbol `STOP` opposed to `stop`? As a constant I'd prefer it to be upper case(though it's not a convention in js).

Reply:
`stop` added